pkg/storages/s3/folder.go

```diff
@@ -319,6 +344,28 @@ func (folder *Folder) partitionToObjects(keys []string) []*s3.ObjectIdentifier {
 	return objects
 }
 
+func (folder *Folder) getUploadTimeout() int {
```

Comment:
It would be more convenient to create `getUploadContext` method, returning `context.Context`
Then we could get rid of `if`’s in all methods.

But see another comments first.